### ISSUE TITLE
Parameter state not allowed on access token requests for Google API

### ISSUE THREAD

**Issue by jasonwcfan** [a85/POSTMan-Chrome-Extension#2811]
1. Postman Version: 4.10.3
 2. App (Chrome app or Mac app): Mac
 3. OS details: MacOS Sierra 10.12.1
 4. Is the Interceptor on and enabled in the app: No
 5. Did you encounter this recently, or has this bug always been there: Don't know, first time using the access token helper
 6. Expected behaviour: After authenticating with google, get back an access token
 7. Console logs (http://blog.getpostman.com/2014/01/27/enabling-chrome-developer-tools-inside-postman/ for the Chrome App, View->Toggle Dev Tools for the Mac app): none
 8. Screenshots (if applicable)
![](http://i.imgur.com/Xncn5M7.png)
![](http://i.imgur.com/VlwZdb3.png)

After making an access token request to https://accounts.google.com/o/oauth2/auth and authenticating in the window, it appears to return a token successfully. However the token can't be used, and in its description mentions that an error occurred with a state parameter that is not allowed

Possible that this is related to #2500 and #2677

**Comment by Wingedillidan** [comment]
I'm also running into this issue on version 5.0.0, except a token doesn't even get generated in the list, it just returns an error 400 in the developer console.

![screen shot 2017-06-13 at 5 10 38 pm](https://user-images.githubusercontent.com/13971542/27104774-420e796a-505b-11e7-9b3a-c669968488d0.png)

**Comment by moneal** [comment]
I'm getting the exact same error with 5.0.0 also. It appears Google doesn't like the state parameter even though its valid.

**Comment by EdwardPerello** [comment]
Also getting this.

**Comment by eldadts** [comment]
also getting this

**Comment by NickBrooks** [comment]
+1

**Comment by waffe7** [comment]
+1 on 4.10.7

**Comment by michalhudecek** [comment]
+1

**Comment by xhanrot** [comment]
+1 on 5.1.3

**Comment by hvkale** [comment]
Is there any way to see the returned token?

**Comment by rossjha** [comment]
+1 on 5.2.0

**Comment by davingreen** [comment]
+1 on v5.2.0 Windows x64
Can we just have an option to not include state parameter?

**Comment by Dave76** [comment]
+1 v5.2.0 on Windows x64. Maybe it would be better to just let us define the entire auth and token requests?

**Comment by joehanko** [comment]
@jasonwcfan just contacted Postman support and one of their amazing engineers actually [documented](https://documenter.getpostman.com/view/2542471/slack-weekly-pageview-report/6thxLUw) this completely. 

### tl;dr Try changing your Auth URL and Access Token URL to 3 & 4 below.

Assuming you've already enabled the proper credentials in Google APIs console, added the correct callback URL from Postman here are the important parts:

1. Choose **OAuth 2.0** from the Type dropdown list
2. Click **Get New Access Token** 
3. Fill Auth URL with `https://accounts.google.com/o/oauth2/v2/auth?access_type=offline `
4. Fill Access Token URL with `https://www.googleapis.com/oauth2/v4/token` 
5. Fill Client ID 
6. Fill Client Secret
7. Fill Scope with `https://www.googleapis.com/auth/analytics.readonly `
8. Select `Authorization Code` from the dropdown list 
9. Check `Request access token locally `
10. Click `Request Token`. [Read More](https://documenter.getpostman.com/view/2542471/slack-weekly-pageview-report/6thxLUw)

I suspect you may have the wrong **Auth URL** and **Access Token URL**, because I used the given ones in Google's Documentation and had the same issue - and have pretty much the same settings and environment you do. I'm not exactly sure why the access type must be specified as offline, but I think it has something to do with the fact that Google APIs will only pass you back a refresh token if you request offline access.

> When you request Offline Access the Google Authentication server returns a Refresh-Token. Refresh tokens give your application the ability to request data on behalf of the user when the user is not present and in front of your application. 

> If a refresh token is present in the authorization code exchange, then it can be used to obtain new access tokens at any time. This is called offline access, because the user does not have to be present at the browser when the application obtains a new access token. [Read More](https://stackoverflow.com/questions/30637984/what-does-offline-access-in-oauth-mean)

You'll also need to save the Access Token and Refresh Tokens in your Postman Environment as variables `access_token` and `bearer_token` with the values that were returned to you when you exchange your creds - "Get New Access Token > Use Token". Once you have the Environment Variables saved, just add a simple Key Value in Headers:
Key: `Authorization` Value `Bearer {{access_token}}`

How to do this: [Dynamically assigning environment variables](https://www.getpostman.com/docs/postman/environments_and_globals/variables)

Something that also might be helpful to you - I've been using the RAW JSON to input - I'd really recommend using the GA [Request Composer](https://ga-dev-tools.appspot.com/request-composer/) for this. Huge thanks to Postman Engineering & Customer Service  for being amazing though. Hope your project hasn't moved on ;)

Cheers & hope this helps guys,
JH

**Comment by harryi3t** [comment]
@joehanko Thanks a ton Joe 👍 
@jasonwcfan Could you confirm if this works for you?

**Comment by harryi3t** [comment]
Hi, everyone

Postman 5.3.0 is out with an extra field for `state` as requested.
Now you can just leave the field empty and Google will stop complaining about it

![postman](https://user-images.githubusercontent.com/5207331/31329355-547ee10e-acf7-11e7-9b77-78e69dbfdffa.png)

I have tested this, can someone confirm that it works?

Thanks
